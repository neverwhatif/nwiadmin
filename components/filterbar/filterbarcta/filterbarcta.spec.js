import React from 'react';
import { shallow } from 'enzyme';

import FilterBarCta from './index';

jest.mock('nwiadmin/components/button', () => 'Button');

const mockData = [{
    label: 'Lorem',
    action: () => jest.fn(),
}, {
    label: 'Ipsum',
    action: () => jest.fn(),
}];

describe('FilterBarCta', () => {
    const component = shallow(<FilterBarCta data={mockData} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
});
