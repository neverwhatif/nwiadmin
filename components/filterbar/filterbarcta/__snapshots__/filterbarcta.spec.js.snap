// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBarCta should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBarCta
    data={
      Array [
        Object {
          "action": [Function],
          "label": "Lorem",
        },
        Object {
          "action": [Function],
          "label": "Ipsum",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="item"
        >
          <Button
            isDisabled={false}
            onClick={[Function]}
          >
            Lorem
          </Button>
        </li>,
        <li
          className="item"
        >
          <Button
            isDisabled={false}
            onClick={[Function]}
          >
            Ipsum
          </Button>
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Lorem",
        "nodeType": "host",
        "props": Object {
          "children": <Button
            isDisabled={false}
            onClick={[Function]}
          >
            Lorem
          </Button>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "isDisabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "Button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "Ipsum",
        "nodeType": "host",
        "props": Object {
          "children": <Button
            isDisabled={false}
            onClick={[Function]}
          >
            Ipsum
          </Button>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ipsum",
            "isDisabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Ipsum",
          "type": "Button",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item"
          >
            <Button
              isDisabled={false}
              onClick={[Function]}
            >
              Lorem
            </Button>
          </li>,
          <li
            className="item"
          >
            <Button
              isDisabled={false}
              onClick={[Function]}
            >
              Ipsum
            </Button>
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Lorem",
          "nodeType": "host",
          "props": Object {
            "children": <Button
              isDisabled={false}
              onClick={[Function]}
            >
              Lorem
            </Button>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem",
              "isDisabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Lorem",
            "type": "Button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "Ipsum",
          "nodeType": "host",
          "props": Object {
            "children": <Button
              isDisabled={false}
              onClick={[Function]}
            >
              Ipsum
            </Button>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ipsum",
              "isDisabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Ipsum",
            "type": "Button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
